### DISCUSSION TITLE
How to remove a reviewer from a Pull Request

### DISCUSSION THREAD

**Discussion by jimn-tn** [community/community#23054] [New to GitHub] [answered]
<p>We have our PR’s to require a minimum amount of reviewers to be approved. Sometimes people are out or priorites shift so others may pick up a the process of reviewing which leaves the previous reviewers on the request. Also, as people comment or get mentioned in the conversation, they get added to the list as well.</p>
<p>Is there a way to totally remove them as a reviewer? I know we can just change their status but visually it’s not the best looking at possibly a list of people who may never actually review something</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/jimn-tn">@jimn-tn</a>,</p>
<p>Thanks for being here! You can dismiss pull request reviews that are no longer valid or are unable to be approved by the reviewer, but as you’ve mentioned on your post their names remain on the list. </p>
<p>We’re always working to improve GitHub and we consider every suggestion we receive. I don’t show that we are planning to implement any flow changes like that to the PR reviews feature in the immediate future, but I have added this to our internal request list for the team to review and consider. We can’t say if/when we may add a feature, but we really appreciate you providing feedback on how we can make GitHub work better for you!</p>
<p>Best,</p>
<p>Andrea</p>

**Reply by lordzsolt** [reply]
5 years later....

**Reply by jordanbtucker** [reply]
I can't believe there is nothing I can do to remove this.

![image](https://github.com/community/community/assets/468153/42d748eb-8922-43ea-9ddf-fa272e61164e)

**Reply by amirna2** [reply]
Yeah..should be a dead simple feature to add. Gitlab has had that from pretty much day 1.

**Reply by Electro-Factory** [reply]
wow

**Reply by dgl2000** [reply]
2025 now

**Reply by daniel-enqz** [reply]
Hello I was able to remove them by clicking this gear icon:
![Screenshot 2025-02-13 at 10 27 02 a m](https://github.com/user-attachments/assets/f9992977-523b-4c07-a9f1-4422c9b12bdf)

it was simple.

**Reply by justin-octo** [reply]
Does this means they finally fixed it?

**Reply by justin-octo** [reply]
Looks like they fixed it.  Works for me now.  I was able to add and then remove a reviewer. :)

**Reply by diegofloresbcp** [reply]
doesnt work for me, I get a 404 error when trying to remove a reviewer

**Reply by nathaniel-flick-ctct** [reply]
Copilot reviewer shows in the dropdown as unchecked, and yet is still a reviewer on the pull request. I cannot remove it, and in fact when I followed its directions it screwed up my pr which had to then go back and fix.

**Comment by mulkieran** [comment]
<p>This would help me, too. If I’m essentially mentoring a new contributor through a PR, I’ld like to remove myself from the list of reviewers when the PR is ready and goes up for actual review. As it stands, I can’t.</p>

**Comment by noamhacker** [comment]
<p>click on the gear icon where you’d normally add reviewers, locate yourself and uncheck to remove yourself</p>

**Reply by ErroneousFatality** [reply]
Doesn't work.

**Reply by Bus42** [reply]
Confirmed. No option to remove anyone you accidentally clicked on because you sneeze.

**Reply by justin-octo** [reply]
Works for me now... I just added a reviewer and then removed him.

**Reply by gollex** [reply]
does not work with firefox.

**Reply by deekshith-sclera** [reply]
It auto adds the default reviewers , however once PR has been raised , we can modify the list of reviewers

**Comment by jusjusjus** [comment]
<p>After clicking on my icon I still appear on the list of Reviewers, but without the yellow dot.  This is not what I am aiming for.</p>

**Comment by demisx** [comment]
<p>I often get assigned to some PRs which I don’t need to be a reviewer anymore. I would like a simple functionality to remove myself from the reviewer list similar to how I can add myself to the list. Currently, click on my name does nothing. Please add this feature.</p>

**Comment by BigLep** [comment]
<p>Any update on this?  I want to allow someone who has interacted with a PR in the past to remove themselves from the reviewer list.  My reasons for this are:</p>
<ol>
<li>Make it clear to the PR author that the reviewer is no longer engaged.</li>
<li>Make my team’s github project board more clear about what PR’s are truly awaiting a certain individual’s review.  For example, we have some longstanding PRs that we’re trying to close down but have assigned new reviewers to them.  These old items are cluttering up for the previous reviewers.</li>
</ol>

**Comment by queenofcorgis** [comment]
<p>Hello <a class="mention" href="/biglep">@BigLep</a> - and thank you for this feedback! It looks like since this topic was initially posted our feedback and feature request process has changed.</p>
<p>If you wouldn’t mind, I encourage you to submit your feedback via the new GitHub <a href="https://github.com/github/feedback/discussions">Feedback Discussions forum</a> so that our product team can better track your request :slight_smile:. It’s a new thing, but we’re trying to push all product feedback to the same new area so it’s easier to find and consolidate user feedback!</p>

**Comment by loic-sharma** [comment]
<p>Hello, I also need this feature to remove stale items from my “Review Requests” page.</p>
<p>I created a GitHub Feedback discussion for this feature request: <a href="https://github.com/github/feedback/discussions/16730" class="inline-onebox" rel="noopener nofollow ugc">Remove myself as reviewer from pull requests · Discussion #16730 · github/feedback · GitHub</a></p>

**Comment by ym-herby** [comment]
<p>This would be super helpful as I have contractors on my projects who are no longer with the company and I’d like to remove them as reviewers.</p>

**Reply by christopherallanperry** [reply]
Same here. A colleague who has provided feedback on a PR has now left the organisation, and I'd like to remove them from the reviewer list.

**Comment by zbynekwinkler** [comment]
I just successfully deleted a reviewer from a pull request using [Github API](https://docs.github.com/en/rest/pulls/review-requests#remove-requested-reviewers-from-a-pull-request).

**Comment by justin-octo** [comment]
Conversly, I accidentally removed all the reviewers by clicking the link next to one to request a renewed review.  It clear ALL the reviewers and just left him.  I have no way to re-add those reviewers.

**Comment by tinoh9** [comment]
I just got the same issue like @justin-octo, don't know how to re-add previous reviewers.

**Comment by justin-octo** [comment]
This seems like such a simple but important function.  Makes me wonder about the future of GitHub if a ticket like this can sit here for 4 1/2 years with little/no effort.

**Comment by bajaj6** [comment]
+1 very imp for solid code review process.

**Comment by parberge** [comment]
So should we vote on https://github.com/orgs/community/discussions/23054 for this?

**Comment by amirna2** [comment]
Yeah. I had a case where someone was able to inject themselves as reviewer on a PR and then out of the blue approved a PR.
I can't either remove them or even block them.

**Comment by justin-octo** [comment]
This is more of a work around, and maybe only really applicable for organizations.  But you can make a Team and assign the Team as a reviewer.  Then as people come and go they are added or removed from the Team.

**Reply by justin-octo** [reply]
Maybe this is why they haven't worked on this.  They want everyone to pay to have an organization for this "feature" instead of giving it away for free.

**Reply by amirna2** [reply]
You can also create a CODEOWNERS file in your `.github` folder where you can specify who can be a reviewer. It's not an idea solution if you have many different roles on a projects.

**Comment by thoraage** [comment, marked as answer]
I just tested and it seems you can remove reviewers with the [github cli](https://cli.github.com/) like this:
```
gh pr edit --repo <your repo> <pr number> --remove-reviewer <login of reviewer>
```

**Reply by JBaczuk** [reply]
So it's basically implemented they just need to add a button on the frontend...

**Reply by jordanbtucker** [reply]
So, if it's available via the CLI, then it's probably also available via the REST API, which means it might be able to be implemented by [Refined GitHub](https://github.com/refined-github/refined-github)?

**Reply by jordanbtucker** [reply]
I just tried this, and it did not work.

**Reply by kerthcet** [reply]
Seems gh also not work .. Have no idea just me or not.

**Reply by kerthcet** [reply]
So I can add reviewer via `gh pr edit $PR_NUMBER --add-reviewer <xxx>`, however, I can't remove reviewer via `gh pr edit $PR_NUMBER --remove-reviewer <xxx>`, but from the front side, I can assign/unassign the reviewers from the reviewer list, if you have the write role.

**Reply by thoraage** [reply]
Is the reviewer a code owner? I've seen that in that case you can't remove them. There might be other right evaluation involved here also; do you have the authority needed to remove them

**Reply by kerthcet** [reply]
From the front side, I mean on the github page, the user can assign/unassign the reviewers from the list. So i think the user has the permission (but not a code owner). 

But with gh tool, specially in workflow, it doesn't work even with exit code 0 returned.

**Reply by Drew-Daniels** [reply]
Noticed the same thing as @kerthcet on my end. I can add reviewers with the CLI, but I cannot remove them with it. I can however remove them using the PR UI.

I've tried this:

```bash

gh pr edit <PR-NUMBER> --remove-reviewer=<COMMA-SEPARATED-LIST-OF-REVIEWER-USERNAMES>
```

**Reply by justin-octo** [reply]
You can now click on the gear icon and remove Reviewers.  They fixed it in the GUI.  Oct 11, 2018 until...

**Reply by kerthcet** [reply]
Is the gh command works as well?

**Comment by Nehal-Mane** [comment]
Request the Reviewer to Remove Themselves

-This is the most straightforward approach. You can politely ask the reviewer to remove themselves from the pull request if they are no longer needed or available for review.
-You can do this by replying to the pull request conversation or sending them a direct message.

**Comment by Nehal-Mane** [comment]
If you have write permissions on the repository (such as Maintainer or Project Owner roles), you can reassign reviewers:
Go to the pull request in GitLab.
- In the right sidebar, under the "Reviewers" section, click the gear icon next to the reviewer you want to remove.
- In the dropdown menu, choose "Change reviewers."
- Select the new reviewer(s) or leave it blank if no further review is needed.
- Click "Update reviewers."

**Comment by arudeep15** [comment]
Removing Reviewers Using Command Line Tools
Some platforms provide command-line tools or APIs to manage pull requests. For example, using the GitHub CLI (gh), you can remove a reviewer from a PR as follows:

Open Terminal.
Run the Command:

gh pr review --remove <username> --repo <owner/repo> <pr-number>

**Comment by myluco** [comment]
I just had that problem.
This is what I did:

- Select the Reviewers "button"; the same button you selected to add the reviewer
- Now it should show the list of the already requested reviewers AND the possible reviewers that you can choose
- Select the reviewer that has already been chosen (the name would have the "selected v" mark.
- That is it

**Reply by jordanbtucker** [reply]
That didn't work for me. The reviewer I want to remove does not have a check mark. There is no way to delete the spam comments they left.

**Reply by justin-octo** [reply]
Ugh. That's annoying.  Does the GH CLI option in the answer work for your situation @jordanbtucker ?
```gh pr edit <PR-NUMBER> --remove-reviewer=<COMMA-SEPARATED-LIST-OF-REVIEWER-USERNAMES>```

**Reply by justin-octo** [reply]
Just thinking creatively... and maybe stupid, but can you export it to GitLab, remove the comments, and then import it back to GitHub?
Seems excessive, but might be a solution.

**Reply by justin-octo** [reply]
Seems like they have fixed the GUI now.
Works for me now.

**Reply by jordanbtucker** [reply]
It's still not working for me. The reviewer I want to remove does not have check mark next to their name, but they are still listed as a reviewer. Did the reviewer you were able to remove leave any comments?

**Comment by justin-octo** [comment]
After 6+ years... since Oct 11, 2018 this has finally been fixed.  The gear icon allows you to remove reviewers.  At least for me it did now.

**Reply by Drew-Daniels** [reply]
@justin-octo please for the love of god stop commenting on every thread, you're blowing up my email and I'm sure many other people's

**Comment by gollex** [comment]
not working in firefox

**Comment by JW-akanm** [comment]
Still broken. Can't remove myself from a previously reviewed PR.
Such a simple feature.

**Comment by t2gran** [comment]
I am working for an open-source project, we require 2 approvals from reviewers, and these are assigned depending who is the best people to do it. I addition a many people may comment or review, all of these end up in the "reviewers" and we lose track of who is assigned to do the review. Our workaround is to use the "assigned" instead - this is fare form a good solution. Please, fix this. Allow us to configure auto-adding reviewers, on/off. Leaving a comment on a PR is NOT a review.

**Comment by chrisdel101** [comment]
**TLDR;** 

Remove the pending review requests 

`gh api -X DELETE /repos/{repo}/pulls/237/requested_reviewers -f reviewers[]={reviewer-username}`
Browser cache clear required.

More...

Ive spend quite a bit of time researching this using the UI I've not been unable to figure it out. Maybe we're all just "doing it wrong"?

The top of the page cli command fails for me without any error.  So using AI (sigh) I was able to get something to happen, well it at least removed the pending review requests I had been assigned, which is better than nothing. 

`gh api /repos/{repo-name}/pulls/{pr-num}` shows all the info and has a `requested_reviewers` block.
`gh api /repos/{repo-name}/pulls/{pr-num}/requested_reviewers` shows just that block's info

The DELETE here doesn't delete the repo. _Should_ just be the reviewers - it worked that way for me anyway.

`gh api -X DELETE /repos/{repo}/pulls/237/requested_reviewers -f reviewers[]={reviewer-username}` - calls the requested_reviewers with DELETE request for the passed in reviewers

**Comment by sovetski** [comment]
Really GitHub?? in 2026 we can not do this simple action????????????
